pkg/plugin/response.go

```diff
@@ -419,3 +436,57 @@ func (r *Response) getDataFrames() (data.Frames, error) {
 		return nil, fmt.Errorf("unknown result type %q", r.Data.ResultType)
 	}
 }
+
+// Hit represents a single hit from the query
+type Hit struct {
+	Fields     map[string]string `json:"fields"`
+	Timestamps []string          `json:"timestamps"`
+	Values     []float64         `json:"values"`
+	Total      int               `json:"total"`
+}
+
+// HitsResponse represents response from the hits query
+type HitsResponse struct {
+	Hits []Hit `json:"hits"`
+}
+
+func (hr *HitsResponse) getDataFrames() (data.Frames, error) {
+	frames := make(data.Frames, len(hr.Hits))
+	for i, hit := range hr.Hits {
+		timeFd := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
```

Comment:
would be great to actually use 2nd param here as `len(hit.Timestamps)`

Reply:
The fix was incorrect. When pre-allocating frames via second param you should use `field.Set(position, value)` instead of `field.Append(value)`. Otherwise, you'll double the amount of fields in result. I wonder why tests didn't catch this change.

Reply:
Oh, they actually did catch this :)

Reply:
fixed, tests, catch this problem